segmentstore/server/host/src/main/java/io/pravega/service/server/host/stat/AutoScaleProcessor.java

```diff
@@ -185,18 +186,29 @@ private void triggerScaleDown(String streamSegmentName, boolean silent) {
 
     private CompletableFuture<Void> writeRequest(ScaleEvent event) {
         CompletableFuture<Void> result = new CompletableFuture<>();
+        CompletableFuture<Void> writeComplete = new CompletableFuture<>();
         try {
-            CompletableFuture.runAsync(() -> {
+            AckFuture ackFuture = writer.get().writeEvent(event.getKey(), event);
+            ackFuture.addListener(() -> writeComplete.complete(null), executor);
+            writeComplete.thenAcceptAsync((Void v) -> {
```

Comment:
sorry, didnt get this comment?
we take the ack future and convert it into a completable future by taking a new future and adding a listener to ackfuture where the listener completes the completable future.
then we chain on this completable future and based on ack future's result, we complete the result future.